src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/AppleCertificatePal.cs

```diff
@@ -399,7 +399,21 @@ private void EnsureCertData()
                 return;
 
             Debug.Assert(!_certHandle.IsInvalid);
-            _certData = new CertificateData(Interop.AppleCrypto.X509GetRawData(_certHandle));
+            string? subjectSummary = Interop.AppleCrypto.X509GetSubjectSummary(_certHandle);
+
+            try
+            {
+                _certData = new CertificateData(Interop.AppleCrypto.X509GetRawData(_certHandle));
+            }
+            catch (CryptographicException e) when (subjectSummary != null)
+            {
+                string message = SR.Format(
+                SR.Cryptography_X509_CertificateCorrupted,
+                subjectSummary);
```

Comment:
Proper indentation, please

```suggestion
                string message = SR.Format(
                    SR.Cryptography_X509_CertificateCorrupted,
                    subjectSummary);
```